openpgp/write.go

```diff
@@ -68,6 +68,7 @@ func detachSign(w io.Writer, signer *Entity, message io.Reader, sigType packet.S
 	}
 
 	sig := new(packet.Signature)
+	sig.MatchIssuerVersion(&signer.PrivateKey.PublicKey)
```

Comment:
I think this one isn't needed, if we already do so in `Signature.Sign`? (And the same thing in key_generation.go.)